pkg/i2gw/providers/gce/gce_extensions_test.go

```diff
@@ -0,0 +1,192 @@
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							betaBackendConfigKey: `{"default":"backendconfig-1"}`,
+						},
+					},
+				},
+			},
+			expectedServices: expectedServices,
+		},
+		{
+			desc: "Specify BackendConfig with beta.cloud.google.com/backend-config annotation, using different BackendConfigs for each port, service will be associated with the BackendConfig for the alphabetically smallest port",
+			services: map[types.NamespacedName]*apiv1.Service{
+				{Namespace: testNamespace, Name: testServiceName}: {
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      testServiceName,
+						Namespace: testNamespace,
+						Annotations: map[string]string{
+							backendConfigKey: `{"ports": {"port1": "backendconfig-1", "port2": "backendconfig-2"}}`,
+						},
+					},
+				},
+			},
+			expectedServices: expectedServices,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			provider := NewProvider(&i2gw.ProviderConf{})
+			gceProvider := provider.(*Provider)
+			gceProvider.storage = newResourcesStorage()
+			gceProvider.storage.Services = tc.services
+			gceProvider.storage.BackendConfigs = backendConfigs
+
+			beConfigToSvcs := getBackendConfigMapping(context.TODO(), gceProvider.storage)
+			backendConfigKey := types.NamespacedName{Namespace: testNamespace, Name: testBeConfigName1}
+			gotServiceList := beConfigToSvcs[backendConfigKey]
+			if len(gotServiceList) != len(tc.expectedServices) {
+				t.Errorf("Got BackendConfig mapped to %d services, expected %d", len(gotServiceList), len(tc.expectedServices))
+			}
+			if !reflect.DeepEqual(gotServiceList, tc.expectedServices) {
+				t.Errorf("Got BackendConfig mapped to %v, expected %v", gotServiceList, tc.expectedServices)
+			}
+		})
+	}
+}
+
+func TestGetBackendConfigName(t *testing.T) {
+	t.Parallel()
+
+	testNamespace := "test-namespace"
+	testServiceName := "test-service"
+	testBeConfigName := "backendconfig-1"
+
+	testCases := []struct {
+		desc           string
+		service        *apiv1.Service
+		beConfigKey    string
+		expectedName   string
+		expectedExists bool
+	}{
+		{
+			desc: "Service using cloud.google.com/backend-config, using default Config over all ports",
+			service: &apiv1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testServiceName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						backendConfigKey: `{"default":"backendconfig-1"}`,
+					},
+				},
+			},
+			beConfigKey:    backendConfigKey,
+			expectedName:   testBeConfigName,
+			expectedExists: true,
+		},
+		{
+			desc: "Service using beta.cloud.google.com/backend-config annotation, using default Config over all ports",
+			service: &apiv1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testServiceName,
+					Namespace: testNamespace,
+					Annotations: map[string]string{
+						betaBackendConfigKey: `{"default":"backendconfig-1"}`,
```

Comment:
Same comment as above - should have both new and old annotation

Reply:
Added the test cases, thanks!